codejail/jail.py

```diff
@@ -0,0 +1,298 @@
+"""
+Core Jail class and supporting code
+"""
+
+from collections import namedtuple
+import json
+import logging
+import os
+import shutil
+
+from .exceptions import JailError, SafeExecException
+from . import languages
+from . import limits
+from .proxy import run_subprocess_through_proxy
+from .subproc import run_subprocess
+from .util import json_safe, temp_directory
+
+log = logging.getLogger("codejail")
+
+
+JailResult = namedtuple('JailResult', ['status', 'stdout', 'stderr'])
+
+
+class Jail(object):
+    """
+    Self contained codejail context.
+    """
+    default_language = languages.other
+
+    known_languages = {
+        'python': languages.python2,
+    }
+
+    def __init__(self, command, bin_path, user, lang=None):
+        self.command = command
+        self.bin_path = bin_path
+        self.user = user
+        if lang is None:
+            self.lang = self.known_languages.get(command, self.default_language)
```

Comment:
The default language (`other`) is designed to be pretty generic.  It adds no command line arguments, and doesn't run under safe_exec, which is the same behavior you would get in the previous version specifying any codejail with a command name other than `"python"`.

In the previous version of codejail, you could do the following:

    >>> from codejail import jail_code
    >>> jail_code.configure('ruby', '/usr/bin/ruby', user='sandbox')
    >>> jail_code.jail_code('ruby', 'puts "hi"')
    ('hi', None, 0)

The default language maintains this behavior, using the `languages.other` language object.  Without the default, you would have to explicitly specify `languages.other`:

    >>> from codejail import jail_code, languages
    >>> jail_code.configure('ruby', '/usr/bin/ruby', user='sandbox')
    <ERROR>
    >>> jail_code.configure('ruby', '/usr/bin/ruby', user='sandbox', lang=languages.other)
    >>> jail_code.jail_code('ruby', 'puts "hi"')
    JailResult(stdout='hi', stderr=None, status=0)

I agree that it's worth expanding the class level docstring, but I'm not sure about breaking backwards compatibility to require the language argument.